nsdb-common/src/main/scala/io/radicalbit/nsdb/common/statement/SQLStatement.scala

```diff
@@ -165,17 +152,41 @@ case object LessOrEqualToOperator    extends ComparisonOperator
 @JsonDeserialize(using = classOf[AggregationJsonDeserializer])
 sealed trait Aggregation
 
+/**
+  * Aggregation that can be applied without a group by clause.
+  */
 @JsonSerialize(using = classOf[GlobalAggregationJsonSerializer])
 @JsonDeserialize(using = classOf[GlobalAggregationJsonDeserializer])
 sealed trait GlobalAggregation extends Aggregation
 
-case object CountAggregation extends GlobalAggregation
-case object MaxAggregation   extends Aggregation
-case object MinAggregation   extends Aggregation
-case object FirstAggregation extends Aggregation
-case object LastAggregation  extends Aggregation
-case object SumAggregation   extends Aggregation
-case object AvgAggregation   extends Aggregation
+/**
+  * Aggregation that is not derived from others.
+  * e.g. average is derived from count and sum.
+  */
+@JsonSerialize(using = classOf[PrimaryAggregationJsonSerializer])
+@JsonDeserialize(using = classOf[PrimaryAggregationJsonDeserializer])
+sealed trait PrimaryAggregation extends Aggregation
+
+/**
+  * Aggregation that is not derived from others.
```

Comment:
one "not" too many

Reply:
Thanks for the hint. It was a bad copy and paste